apis/filesystem/src/test/java/org/jclouds/filesystem/FilesystemBlobStoreTest.java

```diff
@@ -182,6 +182,25 @@ public void testList_NoOptionSingleContainer() throws IOException {
         checkForContainerContent(CONTAINER_NAME, blobsExpected);
     }
 
+    public void testList_EmptyOptionSingleContainer() throws IOException {
+       blobStore.createContainerInLocation(null, CONTAINER_NAME);
+
+       checkForContainerContent(CONTAINER_NAME, null);
+
+       TestUtils.createBlobsInContainer(CONTAINER_NAME, "a", "b", "c");
+       // Test listing where we set the prefix and delimiter to empty string
+       ListContainerOptions options = ListContainerOptions.Builder.delimiter("")
+             .prefix("").afterMarker("");
+       PageSet<? extends StorageMetadata> rs = blobStore.list(CONTAINER_NAME, options);
+       assertEquals(rs.size(), 3);
+       Set<String> expected = Sets.newHashSet("a", "b", "c");
+       for (StorageMetadata sm : rs) {
+          assertTrue(expected.contains(sm.getName()));
+          expected.remove(sm.getName());
```

Comment:
Iterating while removing requires use of `Iterator.remove` to avoid invalidating the iterator.  Instead could you use assertj:

```java
ImmutableList.Builder<String> builder = ImmutableList.builder();
for (StorageMetadata sm : rs) {
    builder.add(sm.getName());
}
assertThat(builder.build()).containsExactly(expected);
```